### PR TITLE
external: restructure the external install steps

### PR DESCRIPTION
**Pull request by parth-gr**
Updating the installation step of external cluster
to make it more user friendly

part-of: https://github.com/rook/rook/discussions/13831

**Checklist:**

- [ ] **Commit Message Formatting**: Commit titles and messages follow guidelines in the [developer guide](https://rook.io/docs/rook/latest/Contributing/development-flow/#commit-structure).
- [ ] Reviewed the developer guide on [Submitting a Pull Request](https://rook.io/docs/rook/latest/Contributing/development-flow/#submitting-a-pull-request)
- [ ] [Pending release notes](https://github.com/rook/rook/blob/master/PendingReleaseNotes.md) updated with breaking and/or notable changes for the next minor release.
- [ ] Documentation has been updated, if necessary.
- [ ] Unit tests have been added, if necessary.
- [ ] Integration tests have been added, if necessary.

### CODE REVIEW

File: deploy/examples/common-external.yaml [hunk 1]
```diff
@@ -18,11 +18,6 @@
 #
 # Then kubectl create -f cluster-external.yaml
 ###################################################################################################################
-apiVersion: v1
```
Comment (subhamkrai) [inline deploy/examples/common-external.yaml:21]: where is the other place we create?

**Comment by travisn** [comment]
The CI is failing when creating the RBAC because of this change. Should the import script instead not create the cluster? IMO we should keep the declarative namespace in the yaml.

**Comment by parth-gr** [comment]
> The CI is failing when creating the RBAC because of this change. Should the import script instead not create the cluster? IMO we should keep the declarative namespace in the yaml.

then we need to change these steps https://rook.io/docs/rook/latest-release/CRDs/Cluster/external-cluster/#import-the-source-data, not sure that we want to do

File: deploy/examples/common-external.yaml (hunk 1 above)
Comment (parth-gr) [reply deploy/examples/common-external.yaml:21]: https://github.com/rook/rook/blob/release-1.13/deploy/examples/import-external-cluster.sh#L71

**Review by subhamkrai — CHANGES_REQUESTED** [review]
@parth-gr could rebase your PR to get the Blaine's change so that MultiCluster CI passes?

**Comment by parth-gr** [comment]
> @parth-gr could rebase your PR to get the Blaine's change so that MultiCluster CI passes?

done

File: Documentation/CRDs/Cluster/external-cluster.md [hunk 2]
```diff
@@ -170,6 +147,29 @@ If not installing with Helm, here are the steps to install with manifests.
...
+
+1. Paste the above output from `create-external-cluster-resources.py` into your current shell to allow importing the source data.
+
+1. The import script in the next step uses the current kubeconfig context by
+   default. If you want to specify the kubernetes cluster to use without
+   changing the current context, you can specify the cluster name by setting
+   the KUBECONTEXT environment variable.
+
+   ```console
+   export KUBECONTEXT=hub-cluster
```
Comment (travisn) [inline Documentation/CRDs/Cluster/external-cluster.md:160]: What is the hub-cluster? I might suggest to remove this example line. Setting an env var is common knowledge IMO.

File: Documentation/CRDs/Cluster/external-cluster.md [hunk 3]
```diff
@@ -170,6 +147,29 @@ If not installing with Helm, here are the steps to install with manifests.
 
 2. Create [common-external.yaml](https://github.com/rook/rook/blob/master/deploy/examples/common-external.yaml) and [cluster-external.yaml](https://github.com/rook/rook/blob/master/deploy/examples/cluster-external.yaml)
 
+### Import the Source Data
```
Comment (travisn) [inline Documentation/CRDs/Cluster/external-cluster.md:150]: This change doesn't affect a duplicate namespace issue. Seems like the purpose of the PR changed? How about updating the PR and commit description with the new proposed update?

**Review by travisn — CHANGES_REQUESTED** [review]

File: Documentation/CRDs/Cluster/external-cluster.md (hunk 2 above)
Comment (parth-gr) [reply Documentation/CRDs/Cluster/external-cluster.md:160]: Mostly we use export in the import script, and thus the change to using KUBECONTEXT, is it fine to update in this PR?

Comment (parth-gr) [reply Documentation/CRDs/Cluster/external-cluster.md]: Updated

**Comment by parth-gr** [comment]
> The commit description still needs updating.

Done

**Review by travisn — APPROVED** [review]

**Review by subhamkrai — APPROVED** [review]

**Pull request merged**